maps/wasteland/market.lua

```diff
@@ -664,49 +655,144 @@ end
-            return 'input'
-        end
-        if inside(ppos, bb) then
-            return 'output'
         end
         return 'none'
     end
-end
 
-local function handle_connected_entity(town_center, market, entity)
-    local mode = get_entity_mode(market, entity)
-    if mode == 'input' then
-        handle_market_input(town_center, market, entity)
+    if inside(entity.drop_position, bb) then
+        return 'input'
     end
-    if mode == 'output' then
-        handle_market_output(town_center, market, entity)
+    if inside(entity.pickup_position, bb) then
+        return 'output'
     end
+    return 'none'
 end
 
-local function on_tick(_)
-    local this = ScenarioTable.get_table()
-    if not this.town_centers then
+local _market_entities_targets = {
+    'burner-inserter',
+    'inserter',
+    'long-handed-inserter',
+    'fast-inserter',
+    'filter-inserter',
+    'stack-inserter',
+    'stack-filter-inserter',
+    'loader',
+    'fast-loader',
+    'express-loader'
+}
+local _long_market_entities_targets = {
+    'long-handed-inserter'
+}
+local market_area_left_top = {0, 0}
+local market_area_right_bottom = {0, 0}
+local market_filter = {
+    area = {left_top = market_area_left_top, right_bottom = market_area_right_bottom},
+    name = _market_entities_targets, force = nil
+}
+local long_market_area_left_top = {0, 0}
+local long_market_area_right_bottom = {0, 0}
+local long_market_filter = {
+    area = {left_top = long_market_area_left_top, right_bottom = long_market_area_right_bottom},
+    name = _long_market_entities_targets, force = nil
+}
+local function on_tick(event)
+    local data = global.tokens.maps_wasteland_table
+    if not data.town_centers then
         return
     end
-    for _, town_center in pairs(this.town_centers) do
-        -- get connected entities on markets
+
+    local is_update_balance_tick = false
+    if event.tick % 60 == 0 then
+        is_update_balance_tick = true
+    end
+
+    local is_find_entities_near_market_tick = false
+    if event.tick % (60 * 8) == 0 then
+        is_find_entities_near_market_tick = true
+    end
+
+    for _, town_center in pairs(data.town_centers) do
         local market = town_center.market
-        local entities = get_connected_entities(market)
-        -- handle connected entity
-        for _, entity in pairs(entities) do
-            if entity.force == market.force then
-                handle_connected_entity(town_center, market, entity)
+        if market.valid then
+            local offers = town_center.offers
+            local force = market.force
+
+            if is_update_balance_tick then
+                local coin_balance = town_center.coin_balance
+                if town_center.prev_coin_balance ~= coin_balance then
+                    rendering.set_text(town_center.coins_text, 'Coins: ' .. coin_balance)
+                    town_center.prev_coin_balance = coin_balance
+                end
+            end
```

Comment:
Is this worth the time gain? Adding prev_coin_balance adds code/state complexity

Reply:
Yes, it takes more time to concatenate a string and there's 1 call at 727.